backend/dataall/modules/s3_datasets_shares/api/queries.py

```diff
@@ -72,10 +26,31 @@
     test_scope='Dataset',
 )
 
+listShareObjects = gql.QueryField(
+    name='listDatasetShareObjects',
+    resolver=list_dataset_share_objects,
+    args=[
+        gql.Argument(name='datasetUri', type=gql.NonNullableType(gql.String)),
+        gql.Argument(name='environmentUri', type=gql.String),
+        gql.Argument(name='page', type=gql.Integer),
+    ],
+    type=gql.Ref('ShareSearchResult'),
+)
 
-getShareLogs = gql.QueryField(
-    name='getShareLogs',
-    args=[gql.Argument(name='shareUri', type=gql.NonNullableType(gql.String))],
-    type=gql.ArrayType(gql.Ref('ShareLog')),
-    resolver=get_share_logs,
+getS3ConsumptionData = gql.QueryField(
+    name='getS3ConsumptionData',
+    args=[gql.Argument(name='shareUri', type=gql.String)],
+    type=gql.Ref('S3ConsumptionData'),
+    resolver=get_s3_consumption_data,
+)  # todo: Once PR 1277 is merged modify shareView to use this query
+
+
```

Comment:
This is the new query used in Worksheets. Instead of listing all shared items and filtering, we just filter the glue databases/tables